### DISCUSSION TITLE
Watch customSelect raw SQL query

### DISCUSSION THREAD

**Discussion by itsJoKr** [simolus3/drift#3042] [Q&A] [answered]
Hey, so I want to use sqlite3, but opted for drift just because I like docs and isolate handling. So I'm using drift but only customSelects and so.

Can I do `customSelect` and watch it? Reading from docs:

> For custom selects, you should use the readsFrom parameter to specify from which tables the query is reading. 

But I don't have any tables, hence I don't know how to use `readsFrom` param since it accepts ResultSetImplementation which from what I see is generated when I use Drift generation. Can I somehow just send my tables as a string?

--- 

From my research, the `readsFrom` will in the end just use table name strings `TableUpdateQuery.onTableName(table)`, but the ResultSetImplementation doesn't let you access this directly. It really wants you to use drift DAO models, which I don't want, I just want to send table name string.

**Comment by itsJoKr** [comment]
This is a dumb workaround that I have for now. I'm creating a table and then never using it but sending it as a parameter to `readsFrom` and `updates`.

```
class DriftSomeModel extends Table {
  TextColumn get id => text()(); // Need to have something here

  @override
  String get tableName => 'DbCacheResponse';
}

```

**Comment by simolus3** [comment, marked as answer]
We're expecting tables instead of names here since these are exposed as getters in a typical drift usage (it would be less convenient to use names which requires users to always use `someTable.tableName`). Adding tables like you did is possible, you can also use a custom table implementation not added to the database (thus not needing any codegen steps):

```dart
class EmptyTable extends Table with TableInfo<EmptyTable, Never> {
  @override
  final String actualTableName;

  EmptyTable(this.actualTableName);

  @override
  List<GeneratedColumn<Object>> get $columns => throw UnimplementedError();

  @override
  DatabaseConnectionUser get attachedDatabase => throw UnimplementedError();

  @override
  TableInfo<EmptyTable, Never> createAlias(String alias) {
    throw UnimplementedError();
  }

  @override
  Never map(Map<String, dynamic> data, {String? tablePrefix}) {
    throw UnimplementedError();
  }
}
```

Then you can use table names by wrapping them in `EmptyTable('DbCacheResponse')` (or write a generic wrapper doing that for you so that you only have to supply table names without wrapping them all individually).

**Reply by itsJoKr** [reply]
Awesome. Thanks for the quick reply.
